### PR TITLE
fix(playback): bound conditional origin chunks

### LINKED ISSUE #138: [bug] Bound conditional origin chunks before buffering response bodies
## Summary

A TestFlight 25 iOS playback report shows bounded direct-stream chunk requests receiving standards-compliant full `200 OK` responses after conditional range validation fails. The current chunk fetcher uses a completion-handler `URLSessionDataTask`, so it buffers the entire response body before it can reject the status. On multi-gigabyte media this can exhaust client memory and stop playback.

## Evidence

- The client requested bounded 4 MiB byte ranges with an entity validator.
- Diagnostic server logs recorded several conditional range requests falling back to full `200` responses between roughly 637 MiB and 3.15 GiB; one response sent about 2.86 GiB.
- A separate request received the expected small `206`, confirming the origin supports ranges.
- A local Foundation probe demonstrated that a continuously paced full response is accumulated until completion; the request timeout is an idle timeout, not a total transfer deadline.
- Memory-pressure termination is the leading explanation, but is not confirmed without a device jetsam report.

## Required behavior

- Validate HTTP status, `Content-Range`, and entity validators when response headers arrive.
- Use `If-Match` for bounded random-access chunks so a stale validator yields a bodyless `412`.
- Reject unexpected nonzero `200` responses before buffering their bodies.
- Cap all accepted media data to the requested interval and cap diagnostic error bodies separately.
- Preserve exact-once request accounting and guarantee no callbacks after cancellation returns.
- Add deterministic simulator coverage for invalid full responses, stale validators, malformed/oversized partial responses, cancellation races, and bounded diagnostic bodies.

## Non-goals

- Treating standard stale `If-Range` behavior as a server error.
- Claiming the server validator mismatch cause without additional evidence.
- Claiming a physical-device jetsam reproduction from simulator-only evidence.

### AI Disclosure
- Tool(s): Codex harness
- Model(s): gpt-5.6-sol
- Involvement: fully AI-generated, human-directed
- Adversarial review: independent agents validated the transport failure and challenged validator and lifecycle assumptions; the linked fix will document how each finding was resolved.

### PR DESCRIPTION
**Pull request by Quick104**
## Problem

Fixes #138.

Companion server contract/diagnostics: Silo-Server/silo-server#594.

An internal TestFlight report showed bounded direct-stream chunk requests receiving full `200 OK` responses after conditional range validation failed. `PlaybackOriginChunkFetcher` used a completion-handler data task, so it could not inspect the status until Foundation had accumulated the entire response in one `Data`. The captured responses were hundreds of MiB to multiple GiB, making memory-pressure termination the likely playback failure.

## Approach

- Move discrete chunk reads to a serial `URLSessionDataDelegate` and validate status, `Content-Range`, and the strong ETag at response headers.
- Send `If-Match` for validator-bound bounded reads so a stale entity receives a bodyless `412`, while leaving the sequential stream's standard `If-Range` behavior unchanged.
- Reject unexpected nonzero `200` responses before their bodies are delivered.
- Bound accepted media to the requested/declared interval and 404 diagnostics to 4 KiB, cancelling transport as soon as the bound is reached.
- Serialize admission, accounting, callbacks, and cancellation so `cancel()` is a callback-quiescence barrier and begin/end counts remain exact.
- Strictly validate partial intervals, including short non-EOF and EOF cases, without losing changed-entity precedence.
- Accept a shorter connection-close-delimited byte-zero `200` only at clean EOF and publish the observed entity length, while retaining the requested-byte streaming cap.

## Testing

- `xcodegen generate`: passed.
- `PlaybackOriginStreamResumeTests`: 31 passed, 0 failed on an iPhone 17 Pro simulator (iOS 26.4).
- Cancellation-race regression repeated 10 times: 10 passed, 0 failed.
- Broader `SiloTests` run before the final focused edge-case additions: 1,255 passed; the same eight unrelated settings/UI tests failed and were reproduced against a clean detached `origin/main` build.
- Final signed iOS simulator build, install, and two terminate/launch cycles: passed.
- Final `SiloTV` simulator build and `SiloMac` arm64 build: passed.
- Final adversarial review: clean; the initial callback-quiescence race, malformed/short `206` handling, and small full-response edge cases were fixed and covered.

The remaining coverage gap is a real-device stream of the reported multi-GiB media. Simulator tests deterministically prove header-time rejection and bounded transport, but cannot prove physical-device jetsam behavior.

### AI Disclosure
- Tool(s): Codex harness
- Model(s): gpt-5.6-sol
- Involvement: fully AI-generated, human-directed
- Adversarial review: independent agents reproduced the transport behavior, rejected an unsafe server-validator change, found cancellation and partial-range edge cases, and verified their fixes in the final diff. The automated PR review found one additional unknown-length EOF case, which is fixed in the follow-up commit and covered by the 31-test focused suite.

### CODE REVIEW

File: iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift [hunk 1]
```diff
@@ -140,161 +191,394 @@ final class PlaybackOriginChunkFetcher {
...
+                .media(
+                    total: total,
+                    entityETag: PlaybackOriginStream.strongETag(
+                        http.value(forHTTPHeaderField: "ETag")
+                    ),
+                    byteLimit: byteLimit
+                ),
+                taskID: taskID
             )
+            return true
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift:339]: Fixed in `710c71c`. A zero-offset `200` with unknown length still streams under the requested-byte cap; on clean EOF it accepts only a nonempty shorter body and publishes the observed byte count as the entity total. Known-length `200` and every `206` path remain exact-only. The new close-delimited regression passes, and the full `PlaybackOriginStreamResumeTests` suite is 31/31.

File: iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift [hunk 2]
```diff
@@ -140,161 +191,394 @@ final class PlaybackOriginChunkFetcher {
...
-                range: range,
-                attempt: attempt,
-                body: body,
-                total: total,
-                responseETag: PlaybackOriginStream.strongETag(
-                    http.value(forHTTPHeaderField: "ETag")
-                )
+            let byteLimit = http.expectedContentLength > 0
+                ? Int(min(Int64(state.range.count), http.expectedContentLength))
+                : state.range.count
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift:327]: Fixed in 710c71c. Clean EOF for a nonempty, close-delimited byte-zero 200 now succeeds while the requested range remains the streaming memory cap; the regression and full resume suite pass.

File: iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift [hunk 3]
```diff
@@ -140,161 +191,394 @@ final class PlaybackOriginChunkFetcher {
...
+            return nil
+        }
+        let unit = trimmed[..<separator]
+        guard unit.lowercased() == "bytes" else { return nil }
+        let value = trimmed[separator...].trimmingCharacters(in: .whitespaces)
+        let slashParts = value.split(separator: "/", omittingEmptySubsequences: false)
+        guard slashParts.count == 2,
+              slashParts[1] != "*",
+              let total = Int64(slashParts[1]),
+              total > 0 else {
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift]: Fixed in fe21f3c. RFC 9110 wildcard totals now parse as unknown and are accepted only for the exact requested interval, since an unknown total cannot prove a shorter interval reached EOF. Simulator coverage and the 33-test resume suite pass.

File: iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift [hunk 4]
```diff
@@ -140,161 +191,394 @@ final class PlaybackOriginChunkFetcher {
...
+    /// Must run on `stateQueue`.
+    private func handleDataLocked(_ data: Data, taskID: Int) -> Bool {
+        guard var state = attempts[taskID], state.response != nil else {
+            return true
+        }
+        switch state.response {
+        case .media(_, _, let byteLimit):
+            let remaining = max(0, byteLimit - state.body.count)
+            if remaining > 0 {
+                state.body.append(contentsOf: data.prefix(remaining))
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift:459]: Fixed in fe21f3c. The serialized delegate path removes the attempt state before appending and reinserts it only when incomplete, so the Data buffer is not shared with the dictionary during mutation. The multi-frame bounded-body tests remain green.

File: iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift [hunk 5]
```diff
@@ -140,161 +191,394 @@ final class PlaybackOriginChunkFetcher {
...
+            return
+        }
+        switch state.response {
+        case .media(_, _, let byteLimit):
+            guard state.body.count == byteLimit else {
+                failLocked(
+                    id: state.id,
+                    range: state.range,
+                    cause: .prematureEOF,
+                    statusCode: nil
```
[the remark this answers is not in the document]
Comment (Quick104) [reply iosApp/iosApp/Screens/Player/PlaybackOriginChunkFetcher.swift]: Fixed in fe21f3c. A clean short 206 now enters the existing prematureEOF retry policy. The new origin regression verifies two identical Range/If-Match requests, one final store, and balanced begin/end accounting; the full resume suite is 33/33.
